tests/ui/waker_clone_and_wake.rs

```diff
@@ -0,0 +1,5 @@
+pub fn wake(cx: &mut std::task::Context) {
+    cx.waker().clone().wake()
+}
```

Comment:
It'd be good to have some test cases for where this shouldn't lint.
- `waker.wake();`
- `waker.clone().wake_by_ref();`

Reply:
I added those tests cases.

But maybe they should probably be linted too !

Especially the second one, which could be part of a larger `useless_clone` lint (I am surprised that it doesn't exist).

Reply:
> Especially the second one, which could be part of a larger `useless_clone` lint (I am surprised that it doesn't exist).

FWIW, this exists: [`redundant_clone`](https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_clone), but it's in the nursery category because of too many false positives and is also undergoing a large rewrite. It looks for a different kind of pattern though (cloning a value and letting the old value drop w/o further use).